preconditions/src/main/java/com/palantir/logsafe/Preconditions.java

```diff
@@ -106,6 +107,18 @@ public static void checkArgument(
         }
     }
 
+    /**
+     * Ensures the truth of an expression involving one or more parameters to the calling method.
+     *
+     * <p>See {@link #checkArgument(boolean, String, Arg...)} for details.
+     */
+    @Contract("false, _, _ -> fail")
+    public static void checkArgument(boolean expression, @CompileTimeConstant String message, List<Arg<?>> args) {
```

Comment:
We also want this to handle `List<SafeArg<?>>`, etc.

```suggestion
    public static void checkArgument(boolean expression, @CompileTimeConstant String message, List<? extends Arg<?>> args) {
```